File: python/codeql/code_injection/benign_edge.py
from flask import Flask, request


app = Flask(__name__)
OPERATIONS = {
    "double": lambda value: value * 2,
    "square": lambda value: value * value,
}


@app.get("/calculate")
def calculate():
    operation = OPERATIONS.get(request.args.get("operation", "double"), OPERATIONS["double"])
    value = int(request.args.get("value", "2"))
    return {"result": operation(value)}

File: javascript/codeql/code_injection/benign_edge.js
const express = require("express");

const app = express();
const operations = new Map([["double", (value) => value * 2], ["square", (value) => value * value]]);

app.get("/calculate", (req, res) => {
  const operation = operations.get(String(req.query.operation || "double")) || operations.get("double");
  res.json({ result: operation(Number(req.query.value || 2)) });
});
